dashboard/modules/reporter/reporter_agent.py

```diff
@@ -348,19 +361,25 @@ def _get_load_avg(self):
         per_cpu_load = tuple((round(x / self._cpu_counts[0], 2) for x in load))
         return load, per_cpu_load
 
+    @staticmethod
+    def _compute_speed_from_hist(hist):
+        while len(hist) > 7:
+            hist.pop(0)
+        then, prev_stats = hist[0]
+        now, now_stats = hist[-1]
+        time_delta = now - then
+        return tuple((y - x) / time_delta for x, y in zip(prev_stats, now_stats))
+
     def _get_all_stats(self):
         now = dashboard_utils.to_posix_time(datetime.datetime.utcnow())
         network_stats = self._get_network_stats()
-
         self._network_stats_hist.append((now, network_stats))
-        self._network_stats_hist = self._network_stats_hist[-7:]
-        then, prev_network_stats = self._network_stats_hist[0]
-        prev_send, prev_recv = prev_network_stats
-        now_send, now_recv = network_stats
-        network_speed_stats = (
-            (now_send - prev_send) / (now - then),
-            (now_recv - prev_recv) / (now - then),
-        )
+        network_speed_stats = self._compute_speed_from_hist(self._network_stats_hist)
+
+        disk_stats = self._get_disk_stats()
```

Comment:
there is only `_get_disk_io` function?

Reply:
Oops, thanks... clearly I did not actually run this :)

Reply:
Can we add some tests?